File: example/scalalib/script/1-use-cases/WebServer.scala
//| mvnDeps: [com.lihaoyi::cask:0.9.1]

object WebServer extends cask.MainRoutes{
  @cask.post("/do-thing")
  def doThing(request: cask.Request) = {
    request.text().reverse
  }

  initialize()
}

File: example/javalib/script/1-use-cases/WebServer.java
//| mvnDeps: [com.sparkjava:spark-core:2.9.4]
import static spark.Spark.*;

public class MinimalApplication {
  public static void main(String[] args) {
    // Define a POST route at /do-thing
    post("/do-thing", (request, response) -> {
      // Reverse the request body text and return it
      String body = request.body();
      return new StringBuilder(body).reverse().toString();
    });
  }
}
